### PR TITLE
React/RN Clients -  Add `useLogout` hook to wrap logout call in mutation

### PR DESCRIPTION
**Pull request by lakardion**
## What this does

Add a hook for logout to wrap it in mutation and use it safely in components.

## Checklist
- [ ] Todo 1
- [ ] Todo 2
- [ ] Todo 3

## How to test

Add user steps to achieve desired functionality for this feature.

### CODE REVIEW

File: {{cookiecutter.project_slug}}/clients/mobile/react-native/src/stores/navigation.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import type { MyNavio } from '@screens/routes'
+import { atom } from 'jotai'
+
```
Comment (paribaker) [inline {{cookiecutter.project_slug}}/clients/mobile/react-native/src/stores/navigation.ts:3]: Where does this get set? It seems to me it is always null

Comment (lakardion) [reply {{cookiecutter.project_slug}}/clients/mobile/react-native/src/stores/navigation.ts:3]: True I think I missed that line in the navio file

**Comment by paribaker** [comment]
@lakardion moved this to in progress as it has conflicts

**Comment by lakardion** [comment]
> @lakardion moved this to in progress as it has conflicts

Done @paribaker

**Review by paribaker — APPROVED** [review]

**Pull request merged**